src/Flysystem/FlysystemAssetStore.php

```diff
@@ -996,4 +997,30 @@ protected function createErrorResponse($code)
 
         return $response;
     }
+
+
+    /**
+     * Find a File object by the given filename.
+     *
+     * @param string $filename Filename to search for, including any custom parent directories.
+     * Should include hashed folder the file is located inside.
+     * @return File
+     */
+    public static function getFileByFilename($filename)
```

Comment:
Perhaps in the function that was removed above you can just replace all the code with a call to this function (and deprecate it)?

Reply:
We'll have a better way of doing this when the permalink job is merged.